sculture-android/app/src/main/java/tr/edu/boun/cmpe/sculture/activity/ProfilePageActivity.java

```diff
@@ -0,0 +1,161 @@
+
+    private int PAGE = 1;
+    private boolean is_loading_more = false;
+    private boolean is_reach_end = false;
+
+    long currentUserID;
+    long visitedUserID;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_profile_page);
+
+
+        loggedInLayout = (RelativeLayout) findViewById(R.id.profile_layout_loggedIn);
+
+        username = (TextView) findViewById(R.id.profile_username);
+        email = (TextView) findViewById(R.id.profile_email);
+
+        follow = (Button) findViewById(R.id.follow);
+
+        story_list_recycler = (RecyclerView) findViewById(R.id.profile_story_list);
+        mLayoutManager = new LinearLayoutManager(this);
+        mStoryListViewAdapter = new StoryListViewAdapter(this);
+        story_list_recycler.setLayoutManager(mLayoutManager);
+        story_list_recycler.setAdapter(mStoryListViewAdapter);
+
+        setRecyclerListeners();
+
+        load_story();
+
+
+        final JSONObject requestBody = new JSONObject();
+        try {
+            currentUserID = requestBody.getLong(PREF_USER_ID);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            JSONObject requestBody2 = new JSONObject(username.getText().toString());
+            visitedUserID = requestBody2.getLong(username.getText().toString());
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        if(currentUserID == visitedUserID)
+            follow.setVisibility(View.GONE);
+
+        follow.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                JSONObject rb = new JSONObject();
+                try {
+                    requestBody.put(API_USER_STORIES, visitedUserID);
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+
+    }
+
+    private void setRecyclerListeners() {
+        story_list_recycler.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                super.onScrollStateChanged(recyclerView, newState);
+                load_story();
+            }
+        });
+    }
+
+    private void load_story() {
+
+        int totalItemCount = mLayoutManager.getItemCount();
+        int lastVisibleIndex = mLayoutManager.findLastVisibleItemPosition();
+
+        boolean loadMore = lastVisibleIndex == totalItemCount - 1;
+
+        if (is_reach_end)
+            Log.i("HERE", "HERE");
+        if ((loadMore && !is_loading_more && !is_reach_end) || PAGE == 1) {
+            is_loading_more = true;
+            JSONObject requestBody = new JSONObject();
+            try {
+                requestBody.put(FIELD_ID, baseApplication.getUSER_ID());
```

Comment:
The RecyclerView is loading the stories of the current logged in user. It should load the stories of the visited user.